core/metadata-3.2.md

```diff
@@ -20,26 +40,139 @@ This proposal aims to codify one mechanism for working with metadata: metadata
+| 771 | `RPL_METADATAUNSUBOK`     | `:<Key1> [<Key2> ...]`                   |
+| 772 | `RPL_METADATASUBS`        | `:<Key1> [<Key2> ...]`                   |
+| 773 | `ERR_METADATATOOMANYSUBS` | `<Key>`                                  |
+| 774 | `ERR_METADATASYNCLATER`   | `<Target> [<RetryAfter>]`                |
+| 775 | `ERR_METADATARATELIMIT`   | `<Target> <Key> <RetryAfter> :<Value>`   |
+
+Reference table listing which subcommands of `METADATA` or any other commands that produce these numerics:
+
+| Label                     | GET | LIST | SET | CLEAR | SUB | UNSUB | SUBS | SYNC | Other   |
+| ------------------------- | --- | ---- | --- | ----- | --- | ----- | ---- | ---- | ------- |
+| `RPL_WHOISKEYVALUE`       |     |      |     |       |     |       |      |      | `WHOIS` |
+| `RPL_KEYVALUE`            | *   | *    | *   | *     |     |       |      |      |         |
+| `RPL_METADATAEND`         |     | *    | *   | *     | *   | *     | *    |      |         |
+| `ERR_METADATALIMIT`       |     |      | *   |       |     |       |      |      |         |
+| `ERR_TARGETINVALID`       | *   | *    | *   | *     | *   | *     | *    | *    |         |
+| `ERR_NOMATCHINGKEY`       | *   |      |     |       |     |       |      |      |         |
+| `ERR_KEYINVALID`          | *   |      | *   |       | *   | *     |      |      |         |
+| `ERR_KEYNOTSET`           |     |      | *   |       |     |       |      |      |         |
+| `ERR_KEYNOPERMISSION`     | *   | *    | *   |       | *   | *     |      |      |         |
+| `RPL_METADATASUBOK`       |     |      |     |       | *   |       |      |      |         |
+| `RPL_METADATAUNSUBOK`     |     |      |     |       |     | *     |      |      |         |
+| `RPL_METADATASUBS`        |     |      |     |       |     |       | *    |      |         |
+| `ERR_METADATATOOMANYSUBS` |     |      |     |       | *   |       |      |      |         |
+| `ERR_METADATASYNCLATER`   |     |      |     |       |     |       |      | *    | `JOIN`  |
+| `ERR_METADATARATELIMIT`   |     |      | *   |       |     |       |      |      |         |
+
+Each subcommand documentation describes the reply and error numerics it expects from the server, but here are brief descriptions of numerics that are used for multiple commands:
+
+Replies:
+
+* `RPL_KEYVALUE` reports the values of metadata keys. The `Visibility` parameter is defined as specified in the [server notification](#todo) format.
+* `RPL_METADATAEND` delimits the end of a sequence of metadata replies.
+
+Errors:
+
+* `ERR_TARGETINVALID` when a client refers to an invalid target.
+* `ERR_KEYINVALID` when a client refers to an invalid key.
+* `ERR_KEYNOPERMISSION` when a client attempts to access or set a key on a target without sufficient permission.
+
+### Keys and values
+
+Keys MUST be restricted to the ranges `A-Z`, `a-z`, `0-9`, and `_.:-` and are case-insensitive. Values are unrestricted, except that they MUST be UTF-8.
```

Comment:
hmm... we should specify that the first character of a key can't be `:`, for obvious reasons.